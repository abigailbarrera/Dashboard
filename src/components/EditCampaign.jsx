import React, { useState, useContext, useEffect } from "react";
import { GlobalContext } from "../context/GlobalState";
import { useHistory, Link } from "react-router-dom";
import styled from 'styled-components';

import { Form, Button, Container, Row } from 'react-bootstrap';

const H1 = styled.h1`
    color: #282c34;
    margin: 0 auto;
    padding: 50px 0;
    font-size: 48px;
    font-size: 48px;
    letter-spacing: -1.5px;
`

export const Editcampaign = route => {
  let history = useHistory();
  const {segments} = React.useContext(GlobalContext);
  const { campaigns, editCampaign } = useContext(GlobalContext);
  const [selectedCampaign, setSeletedCampaign] = useState({
    id: null,
    name: "",
    text: "",
    status: "Preview",
    segment_id: "",
    media: "",
    stats: null
  });
  const currentCampaignId = route.match.params.id;

  useEffect(() => {
    const selectedCampaign = campaigns.find(camp => camp.id === parseInt(currentCampaignId));
    setSeletedCampaign(selectedCampaign);

  }, [currentCampaignId, campaigns]);

  const onSubmit = e => {
    e.preventDefault();
    editCampaign(selectedCampaign);
    history.push("/");
  };

  const handleOnChange = (campaignKey, value) => {
    setSeletedCampaign({ ...selectedCampaign, [campaignKey]: value });
  }

  const handleSelectChange = (campaignKey, value) => {
    const selectedSegment = segments.filter(segment => segment.name === value);
    setSeletedCampaign({ ...selectedCampaign, [campaignKey]: selectedSegment[0].id});
  }

  return (
    <Container>
      <Row>
        <div className="col-6 mt-5">
          <H1>Edit Campaign</H1>
          <Form onSubmit={onSubmit}>
            <Form.Group controlId="formBasicName">
              <Form.Label>Campaign Name</Form.Label>
              <Form.Control
                type="name"
                placeholder="Enter name"
                value={selectedCampaign.name}
                onChange={e => handleOnChange("name", e.target.value)}
              />
            </Form.Group>

            <Form.Group controlId="exampleForm.ControlSelect1">
              <Form.Label>Segment</Form.Label>
              <Form.Control as="select" onChange={e => handleSelectChange("segment_id", e.target.value)}>
                {segments.map(segment => {
                    return (
                      <option key={segment.id}>{segment.name}</option>
                    )
                })}
              </Form.Control>
            </Form.Group>

            <Form.Group controlId="formBasicName">
              <Form.Label>Add Photo or Gif</Form.Label>
              <Form.Control
                type="media"
                placeholder="Enter url"
                value={selectedCampaign.media}
                onChange={e => handleOnChange("media", e.target.value)}
              />
            </Form.Group>

            <Form.Group controlId="exampleForm.ControlTextarea1">
              <Form.Label>Message</Form.Label>
              <Form.Control
                as="textarea"
                rows="3"
                value={selectedCampaign.text}
                onChange={e => handleOnChange("text", e.target.value)}
              />
            </Form.Group>

            <Button variant="primary" type="submit">
              Save
            </Button>

            <Link to="/">
                <button type="button" className="btn btn-link m-5">Cancel</button>
            </Link>
          </Form>
        </div>
      </Row>
    </Container>
  );
};