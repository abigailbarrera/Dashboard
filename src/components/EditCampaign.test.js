import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
 
import { Editcampaign } from './EditCampaign';

const route = {
  match: {
    params: {
      id: "4"
    }
  }
};
 
describe('Editcampaign', () => {
  it('snapshot renders', () => {
    let tree = create(
        <BrowserRouter>
            <Editcampaign {...route} />
        </BrowserRouter>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders header', () => {
    const wrapper = shallow(<Editcampaign {...route} />);
    const header = "Edit Campaign";

    expect(wrapper.contains(header)).toEqual(true);
  });
});
