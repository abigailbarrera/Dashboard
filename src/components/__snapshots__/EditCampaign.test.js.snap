// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editcampaign snapshot renders 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-6 mt-5"
    >
      <h1
        className="sc-AxjAm iNZlfB"
      >
        Edit Campaign
      </h1>
      <form
        className=""
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="formBasicName"
          >
            Campaign Name
          </label>
          <input
            className="form-control"
            id="formBasicName"
            onChange={[Function]}
            placeholder="Enter name"
            type="name"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="exampleForm.ControlSelect1"
          >
            Segment
          </label>
          <select
            className="form-control"
            id="exampleForm.ControlSelect1"
            onChange={[Function]}
          >
            <option>
              All Subscribers
            </option>
            <option>
              Beauty Purchases
            </option>
            <option>
              Decor Purchases
            </option>
          </select>
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="formBasicName"
          >
            Add Photo or Gif
          </label>
          <input
            className="form-control"
            id="formBasicName"
            onChange={[Function]}
            placeholder="Enter url"
            type="media"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="exampleForm.ControlTextarea1"
          >
            Message
          </label>
          <textarea
            className="form-control"
            id="exampleForm.ControlTextarea1"
            onChange={[Function]}
            rows="3"
            value=""
          />
        </div>
        <button
          className="btn btn-primary"
          disabled={false}
          type="submit"
        >
          Save
        </button>
        <a
          href="/"
          onClick={[Function]}
        >
          <button
            className="btn btn-link m-5"
            type="button"
          >
            Cancel
          </button>
        </a>
      </form>
    </div>
  </div>
</div>
`;
